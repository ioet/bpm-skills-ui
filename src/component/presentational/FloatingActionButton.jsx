import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button/Button';
import Tooltip from '@material-ui/core/Tooltip/Tooltip';
import { withStyles } from '@material-ui/core';
import { FabStyles } from '../../styles';
import { TooltipConst } from '../../constants';

const FloatingActionButton = (props) => {
  const {
    classes, tooltip, icon, createSkill,
  } = props;

  return (
    <Tooltip
      title={tooltip}
      placement="left-start"
      enterDelay={TooltipConst.ENTER_DELAY}
      leaveDelay={TooltipConst.LEAVE_DELAY}
    >
      <Button
        variant="fab"
        className={classes.fab}
        color="secondary"
        onClick={(e) => {
          e.preventDefault();
          createSkill();
        }}
      >
        {icon}
      </Button>
    </Tooltip>
  );
};

FloatingActionButton.propTypes = {
  classes: PropTypes.object.isRequired,
  tooltip: PropTypes.string.isRequired,
  icon: PropTypes.element.isRequired,
  createSkill: PropTypes.func.isRequired,
};

export default withStyles(FabStyles)(FloatingActionButton);
