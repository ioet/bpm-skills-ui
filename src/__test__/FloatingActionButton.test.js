import React from 'react';
import FloatingActionButton from '../component/presentational/FloatingActionButton';

describe('<FloatingActionButton />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<FloatingActionButton />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
