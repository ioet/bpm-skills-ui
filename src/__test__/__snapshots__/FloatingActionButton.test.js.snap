// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingActionButton /> Component Renders correctly 1`] = `
<FloatingActionButton
  classes={
    Object {
      "fab": "FloatingActionButton-fab-1",
    }
  }
/>
`;
